### PR TITLE
[FLOC-4312] Port acceptance tests to RHEL 7.2

### PR DESCRIPTION
**Pull request by myechuri**
Not ready for review.

Port acceptance tests to run on RHEL 7.2. Tested standalone on RHEL buildslave reporting to a staged buildmaster.

### CODE REVIEW

File: flocker/provision/_install.py [hunk 1]
```diff
@@ -519,14 +545,18 @@ def task_package_install(package_name, distribution,
...
+            package_distribution = 'centos-7'
         result_path = posixpath.join(
-            '/results/omnibus/', package_source.branch, distribution)
+            '/results/omnibus/', package_source.branch, package_distribution)
         base_url = urljoin(package_source.build_server, result_path)
     else:
         base_url = None
 
-    if is_centos(distribution):
+    if is_centos(distribution) or is_rhel(distribution):
```
Comment (moshez) [inline flocker/provision/_install.py]: We should probably have a function "is_redhatish"?

File: flocker/provision/_install.py [hunk 2]
```diff
@@ -1250,6 +1280,7 @@ def maybe_disable(unit):
 _flocker_uninstallers = {
     "ubuntu-14.04": _uninstall_flocker_ubuntu1404,
     "centos-7": _uninstall_flocker_centos7,
+    "rhel-7.2": _uninstall_flocker_centos7,
```
Comment (moshez) [inline flocker/provision/_install.py]: Why is this not failing on ubuntu 15.10?

File: flocker/provision/_install.py [hunk 3]
```diff
@@ -519,14 +545,18 @@ def task_package_install(package_name, distribution,
     :return: a sequence of commands to run on the distribution
     """
     if package_source.branch:
-        # A development branch has been selected - add its Buildbot repo
```
Comment (moshez) [inline flocker/provision/_install.py]: This function mixes in distro-agnostic code and distro-specific code. Would prefer if it was split out.

File: flocker/provision/_aws.py [hunk 4]
```diff
@@ -36,6 +36,7 @@
     'centos-7': 'centos',
     'ubuntu-14.04': 'ubuntu',
     'ubuntu-15.10': 'ubuntu',
+    'rhel-7.2': 'ec2-user',
```
Comment (moshez) [inline flocker/provision/_aws.py]: There are a lot of places which we need to change to support a new distribution. We should probably localize them into the [Strategy Pattern](https://en.wikipedia.org/wiki/Strategy_pattern)

File: flocker/provision/_install.py (hunk 1 above)
Comment (myechuri) [inline flocker/provision/_install.py]: Fixed.

File: flocker/provision/_install.py (hunk 2 above)
Comment (myechuri) [inline flocker/provision/_install.py]: 15.10 support is still partial (https://docs.clusterhq.com/en/latest/supported/index.html#supported-operating-systems does not list 15.10, and we do not have tests running on 15.10 except for benchmark work). Thanks for pointing out. @wallrj is planning to work on 16.10 support - we will update the thread with 15.10 plan soon.

Comment (myechuri) [inline flocker/provision/_install.py]: Confirming that 15.10 support is no longer required, based on in-person discussions. We only have client support on 15.10 for benchmarking work.

File: flocker/provision/_install.py (hunk 3 above)
Comment (myechuri) [inline flocker/provision/_install.py]: Thanks for suggesting! Fixed.

File: flocker/provision/_aws.py (hunk 4 above)
Comment (myechuri) [inline flocker/provision/_aws.py]: Deferring to [FLOC-4386](https://clusterhq.atlassian.net/browse/FLOC-4386) .

File: flocker/provision/_install.py [hunk 5]
```diff
@@ -95,16 +95,29 @@ def tag_as_test_install(flocker_version, distribution, package_name):
...
+def is_centos_or_rhel(distribution):
+    """
+    Determine whether the named distribution is a version of CentOS or RHEL.
+
+    :param bytes distribution: The name of the distribution to inspect.
+
+    :return: ``True`` if the distribution named is a version of CentOS or RHEL,
+        ``False`` otherwise.
+    """
+    return (distribution.startswith("centos-") or
```
Comment (moshez) [inline flocker/provision/_install.py]: Please use

    distribution.startswith(('censtos-', 'rhel-'))

Comment (myechuri) [inline flocker/provision/_install.py]: Thanks! Updated.

**Pull request merged**